api/v2/api.go

```diff
@@ -508,7 +509,7 @@ func (api *API) getSilencesHandler(params silence_ops.GetSilencesParams) middlew
 	matchers := []*labels.Matcher{}
 	if params.Filter != nil {
 		for _, matcherString := range params.Filter {
-			matcher, err := labels.ParseMatcher(matcherString)
+			matcher, err := compat.Matcher(matcherString)
```

Comment:
Thinking about Mimir here, we might want to add the function pointer to `API` instead of using the global compat here. This would allow us to switch between parsing modes for individual tenants inside of having to do it server-wide.